File: code/dedup_works.R
# Load dependencies
library(readr)

# List TXT files in data directory
fls <- list.files(path = paste0(data_dir, 'working/'),
                    pattern = '*.txt',
                    ignore.case = TRUE)

# De-duplicate results
## Set file count
fl <- 1
## Load and de-duplicate first file
works <- read_delim(file = paste0(data_dir, 'working/', fls[fl]),
                  delim = '|',
                  show_col_types = FALSE)
works <- works[!duplicated(works), ]
## Handle remaining files
while(fl < length(fls)){
  # Increment file count
  fl <- fl + 1
  # Append and de-duplicate next file
  works <- rbind(works,
                 read_delim(file = paste0(data_dir, 'working/', fls[fl]),
                            delim = '|',
                            show_col_types = FALSE))
  works <- works[!duplicated(works), ]
}

# Save results (to file)
write.table(works, 
            file = paste0(data_dir, 'all_works.txt'),
            sep = '|',
            row.names = FALSE)

# Clean up temporary objects
unlink(paste0(data_dir, 'working/'), recursive = TRUE)
rm('fl', 'fls', 'works')
gc()

File: code/dedup_works.py
# -*- coding: utf-8 -*-
"""
Created on Wed Mar  5 11:19:43 2025

@author: kevinatpenn
"""

import os
import pandas as pd
gc

# List TXT files in data directory
working_dir = os.path.join(data_dir, 'working')
fls = [f for f in os.listdir(working_dir) if f.lower().endswith('.txt')]

# De-duplicate results
if fls:
    # Load and de-duplicate first file
    fl = 0
    works = pd.read_csv(os.path.join(working_dir, fls[fl]), sep='|')
    works = works.drop_duplicates()
    
    # Handle remaining files
    while fl < len(fls) - 1:
        fl += 1
        next_df = pd.read_csv(os.path.join(working_dir, fls[fl]), sep='|')
        works = pd.concat([works, next_df], ignore_index=True).drop_duplicates()
    
    # Save results (to file)
    works.to_csv(os.path.join(data_dir, 'all_works.txt'), sep='|', index=False)

# Clean up temporary objects
for file in fls:
    os.remove(os.path.join(working_dir, file))
os.rmdir(working_dir)

del fl, fls, works
gc.collect()
